pkg/kube/services.go

```diff
@@ -199,3 +208,20 @@ func WaitServiceExists(name string, namespace string, cli kubernetes.Interface,
 
 	return svc, err
 }
+
+func createCertificateForService(serviceName string, nameSpace string, hosts string, kubeclient kubernetes.Interface) (*corev1.Secret, error) {
+	caCert, err := kubeclient.CoreV1().Secrets(nameSpace).Get(types.SiteCaServicesSecret, metav1.GetOptions{})
+
+	if err != nil {
+		return nil, err
+	}
+
+	serviceCert := certs.GenerateSecret(serviceName, serviceName, hosts, caCert)
```

Comment:
I think we should prefic the secret name with skupper-, just sp when you look at all the secrets you know which ones were created by skupper.

Also, ultimately not every service will use tls, so we need to think about how that option is selected.

Reply:
Agree on adding the prefix for the secret.

How about adding a flag to the init command to disable tls on http2 connector?

Reply:
Yes, it needs to be per-service. Thinking aloud, the choice of using tls for ingress and egress are in theory orthogonal. For the ingress we probably do want it to be in ServiceBinding so that it can be communicated to other sites so they can also generate a secret for it. Even for egress, if we move to having the service-controller do the generation at some point, then we would want it in ServiceBinding. What if we have tlsCredentials as a string, holding the name of the secret to be used, in both the main ServiceBinding and also in ServiceTarget? (That way we could in future perhaps allow user specified credentials as well as skupper generated ones).

Would then also need to think if the flags for the cli that would enable these options.

Wdyt?

Reply:
Sounds good!

Reply:
Solved except the part of making the tls config optional.

Reply:
By default, services are not going to have TLS config, only when it is specified in with the flag. If the argument for the --tlsCredentials flag is "default", the generated certificates for the service will be used. What do you thing about this?